src/Drawer/Drawer.js

```diff
@@ -92,34 +87,244 @@ class Drawer extends React.Component {
     firstMount: true,
   };
 
-  componentWillReceiveProps() {
+  componentDidMount() {
+    if (this.props.variant === 'temporary') {
+      this.enableSwipeHandling();
+    }
+    this.mounted = true;
+  }
+
+  componentWillReceiveProps(nextProps) {
     this.setState({
       firstMount: false,
     });
+
+    if (this.props.variant !== 'temporary' && nextProps.variant === 'temporary') {
+      this.enableSwipeHandling();
+    } else if (this.props.variant === 'temporary' && nextProps.variant !== 'temporary') {
+      this.disableSwipeHandling();
+    }
+  }
+
+  componentWillUnmount() {
+    this.disableSwipeHandling();
+    this.removeBodyTouchListeners();
+    this.mounted = false;
+  }
+
+  onBodyTouchStart = event => {
+    // sometimes the event handler is called after unbinding it
+    if (!this.mounted) return;
+
+    const anchor = this.getAnchor();
+    const swipeAreaWidth = this.props.swipeAreaWidth;
+
+    const touchStartX =
+      anchor === 'right'
+        ? document.body.offsetWidth - event.touches[0].pageX
+        : event.touches[0].pageX;
+    const touchStartY =
+      anchor === 'bottom'
+        ? window.innerHeight - event.touches[0].clientY
+        : event.touches[0].clientY;
+
+    if (!this.props.open) {
+      if (this.isHorizontalSwiping()) {
+        if (touchStartX > swipeAreaWidth) return;
+      } else if (touchStartY > swipeAreaWidth) return;
+
+      if (this.props.disableSwipeToOpen) return;
+    }
+
+    this.maybeSwiping = true;
+    this.forceUpdate();
+    this.touchStartX = touchStartX;
+    this.touchStartY = touchStartY;
+
+    if (!this.props.open && !this.props.disableAccidentalDiscovery) {
+      this.setPosition(this.getMaxTranslate() - swipeAreaWidth);
+    }
+
+    document.body.addEventListener('touchmove', this.onBodyTouchMove, { passive: false });
+    document.body.addEventListener('touchend', this.onBodyTouchEnd);
+    document.body.addEventListener('touchcancel', this.onBodyTouchEnd);
```

Comment:
What's the use case for this listener? Can't we remove it? (I don't use it on react-swipeable-views side)

Reply:
https://plus.google.com/+PaulIrish/posts/KTwfn1Y2238